src/main/java/edu/kis/powp/jobs2d/command/ExceedingCanvasCommandVisitor.java

```diff
@@ -0,0 +1,57 @@
+package edu.kis.powp.jobs2d.command;
+
+import javax.swing.*;
+import java.awt.Dimension;
+import java.util.Iterator;
+import java.util.logging.Logger;
+
+public class ExceedingCanvasCommandVisitor implements CommandVisitor {
+
+    private JPanel drawArea;
+    private static Logger logger = Logger.getLogger(Logger.GLOBAL_LOGGER_NAME);
+
+    public ExceedingCanvasCommandVisitor(JPanel drawArea) {
+        this.drawArea = drawArea;
+    }
+
+    private Dimension getDrawAreaSize() {
+        return drawArea.getSize();
+    }
+
+    @Override
+    public void visit(OperateToCommand operateToCommand) {
+        Dimension size = getDrawAreaSize();
+        int width = size.width;
+        int height = size.height;
+
+        int x = operateToCommand.getX() + width / 2;
+        int y = height / 2 - operateToCommand.getY();
+
+        if (x >= width || x < 0 || y >= height || y < 0) {
+            logger.warning("Exceeding canvas: (" + x + ", " + y + ")");
```

Comment:
The visitor should not be responsible for a reaction to the result of the analysis. It should only provide a method for checking the result of visitation.